### PR TITLE
[refactor/tests] Added tests for Redis PubSub replication

### PR DESCRIPTION
**Pull request by rennokki**
This PR adds some changes to the PubSub replication PR (https://github.com/beyondcode/laravel-websockets/pull/140) introduced by @francislavoie 

Currently, the tests are set up to run for both Local and Redis clients in the same CI pipeline.

### CODE REVIEW

File: src/WebSocketsServiceProvider.php [hunk 1]
```diff
@@ -56,19 +57,19 @@ public function boot()
 
     protected function configurePubSub()
     {
-        if (config('websockets.replication.enabled') !== true || config('websockets.replication.driver') !== 'redis') {
+        if (config('websockets.replication.driver') === 'local') {
```
Comment (francislavoie) [inline src/WebSocketsServiceProvider.php:60]: This is technically a breaking change - just a reminder to make sure to note that in the v2 release notes that the replication config will need to be added when upgrading. Or you could keep BC by just defaulting to local if no driver is set.

**Comment by francislavoie** [comment]
FWIW I called it "replication" because it's not purely pubsub due to having the channel methods in the interface. But no complaints from me, either is okay.

File: src/WebSocketsServiceProvider.php (hunk 1 above)
Comment (rennokki) [reply src/WebSocketsServiceProvider.php]: Nice catch 🐛

**Comment by rennokki** [comment]
Then I'll just revert the namings. I want something really concise. 🤔 `replicator` ?

**Comment by rennokki** [comment]
Just found out that tests can be done by using mocking.

https://github.com/clue/reactphp-redis/blob/master/tests/FactoryLazyClientTest.php

**Pull request merged**